File: benchmark.py
import hmac
import hashlib
import time

KEY = b"secret_key"
MESSAGE = b"Hello, World!"
ITERATIONS = 1000

start_time = time.time()

for _ in range(ITERATIONS):
    hmac.new(KEY, MESSAGE, hashlib.sha256).digest()

end_time = time.time()

print(f"Python: HMAC operation executed {ITERATIONS} times in {end_time - start_time:.6f} seconds")

File: benchmark.c
#include <stdio.h>
#include <string.h>
#include <openssl/hmac.h>
#include <time.h>

#define KEY "secret_key"
#define MESSAGE "Hello, World!"
#define ITERATIONS 1000

int main() {
    unsigned char hmac[EVP_MAX_MD_SIZE];
    unsigned int hmac_len;
    clock_t start, end;
    double cpu_time_used;

    start = clock();

    for (int i = 0; i < ITERATIONS; i++) {
        HMAC(EVP_sha256(), KEY, strlen(KEY), (unsigned char*)MESSAGE, strlen(MESSAGE), hmac, &hmac_len);
    }

    end = clock();
    cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;

    printf("C: HMAC operation executed %d times in %f seconds\n", ITERATIONS, cpu_time_used);

    return 0;
}